buffer/src/main/java/io/netty/buffer/api/DefaultCompositeBuffer.java

```diff
@@ -447,6 +453,83 @@ public void copyInto(int srcPos, Buffer dest, int destPos, int length) {
         }
     }
 
+    @Override
+    public int readIntoChannelWrite(WritableByteChannel channel, int length) throws IOException {
+        if (!isAccessible()) {
+            throw bufferIsClosed(this);
+        }
+        length = Math.min(readableBytes(), length);
+        checkReadBounds(readerOffset(), length);
```

Comment:
How can this ever throw as you used `Math.min(...)` before ?

Reply:
It can throw if you pass a negative length.